.github/workflows/test.yml

```diff
@@ -59,47 +60,60 @@ jobs:
       PYTHONDEVMODE: '1'
     steps:
       - name: Set up Python
-        uses: actions/setup-python@v5
+        uses: Quansight-Labs/setup-python@v5
         with:
           python-version: ${{ matrix.py }}
           architecture: ${{ matrix.arch }}
 
       - name: Install Rust
-        if: matrix.arch == 'x64'
+        if: matrix.arch == 'x64' && !startsWith(matrix.py, '3.13')
         uses: dtolnay/rust-toolchain@v1
         with:
           toolchain: stable
 
       - uses: actions/checkout@v4
 
       - name: Install Dependencies
+        if: "!endsWith(matrix.py, 't')"
         shell: bash
         run: |
           python -m pip install --require-hashes -r ci/requirements.txt
 
+      - name: Install Dependencies (free-threading)
+        if: "endsWith(matrix.py, 't')"
+        shell: bash
+        run: |
+          python -m pip install --require-hashes -r ci/requirements.freethreading.txt
+
       # TODO enable once PyO3 supports 3.13.
       - name: Build (Rust)
-        if: matrix.arch == 'x64' && matrix.py != '3.13'
+        if: matrix.arch == 'x64' && !startsWith(matrix.py, '3.13')
         run: |
           python -m pip install --config-settings='--build-option=--rust-backend' -e .
 
       - name: Build (No Rust)
-        if: matrix.arch != 'x64' || matrix.py == '3.13'
+        if: matrix.arch != 'x64' || startsWith(matrix.py, '3.13')
         run: |
           python -m pip install -e .
 
       - name: Test C Backend
         run: |
           pytest --numprocesses=auto --hypothesis-profile=${HYPOTHESIS_PROFILE} -v tests/
 
+      - name: Test in Parallel
+        if: "endsWith(matrix.py, 't')"
+        run: |
+          pytest --numprocesses=auto --hypothesis-profile=${HYPOTHESIS_PROFILE} --parallel-threads=10 -v tests/
+
       - name: Test CFFI Backend
+        if: "!startsWith(matrix.py, '3.13')"  # see pyproject.toml:4
```

Comment:
Presumably this limitation no longer holds.

Reply:
Unfortunately no, CFFI is one of the last low-level major dependencies without support. Recently the maintainers asked our team to work on a fork with free-threading support so they can review one big PR:

https://github.com/python-cffi/cffi/pull/143#issuecomment-2580781899